packages/block-library/src/embed/util.js

```diff
@@ -121,7 +132,7 @@ export const createUpgradedEmbedBlock = (
 	// This is not the WordPress embed block so transform it into one.
 	return createBlock( DEFAULT_EMBED_BLOCK, {
 		url,
-		...WP_VARIATION.attributes,
+		...WP_VARIATION?.attributes,
```

Comment:
Even if `?.` prevents type errors, is this still correct when there is no WP_VARIATION?

Reply:
@mcsf is right. In the above check in line 129

```
if ( ! preview || ! isFromWordPress( preview.html ) || isCurrentBlockWP ) {
		return;
}
```
should also check if  `WP_VARIATION` exists and if not it should return nothing so as to not convert to WP block.

```
if ( ! preview  || ! WP_VARIATION  || ! isFromWordPress( preview.html ) || isCurrentBlockWP) {
		return;
}
```

The reason for not suggesting to remove `WP_VARIATION` from `isCurrentBlockWP` variable is that in that check the `providerNameSlug` can be `undefined` in some cases ( called from image block for example ) and that could lead to false positives for a WP block.

Reply:
The conditionals in `createUpgradedEmbedBlock` seem a bit convoluted and hard to read :confused: It seems like if we need `WP_VARIATION` in `isCurrentBlockWP`, but later on need to check for both criteria again, there's some room for improvement. Maybe we can shuffle some of these checks around to move early, and evaluate some stuff only if we have ruled out a few cases :thinking:

Reply:
2f81a0ff7f